.github/workflows/main.yml

```diff
@@ -38,18 +38,21 @@ jobs:
 
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
 
-    - name: Set up python 3.9
-      uses: actions/setup-python@v1
+    - name: Set up python 3.12
+      uses: actions/setup-python@v5
       with:
-        python-version: 3.9
+        python-version: "3.12"
```

Comment:
This is not a general-purpose project: we're only ever deploying dias in one place.  If you've deployed dias in https://github.com/chime-experiment/ch_ansible/pull/575 using 3.11, then the CI should be running in 3.11 as well.

There's no benefit to using different versions in the CI and the deployment.  The CI should be testing against the environment we're using for deployment.

Reply:
That's a property of which anisble version you're using and not related to dias.